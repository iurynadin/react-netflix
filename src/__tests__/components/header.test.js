import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import { Header } from "../../components"
import { BrowserRouter as Router } from 'react-router-dom';
import * as ROUTES from "../../constants/routes";

describe('<Header />', () => {
    it('renders the basic <Header /> with a background', () => {
        const { container, getByText, getByTestId } = render(
            <Header>
                <Header.Frame>
                    <Router>
                        <Header.Logo to="/" alt="Netflix" src="/logo.svg" />
                        <Header.TextLink active="true">Hello I am a link!</Header.TextLink>
                    </Router>
                </Header.Frame>
            </Header>
        );

        expect(getByText('Hello I am a link!')).toBeTruthy();
        expect(getByTestId('header-bg')).toBeTruthy();
        // expect(container.firstChild).toMatchSnapshot();
    });


    it('renders the basic <Header /> without a background', () => {
        const { container, getByText, queryByTestId } = render(
            <Router>
                <Header bg={false}>
                    <Header.Frame>
                        <Header.Logo to={ROUTES.HOME} src="/logo.svg" alt="Netflix" />
                        <Header.ButtonLink to={ROUTES.SIGN_IN}>Sign In</Header.ButtonLink>
                        <Header.TextLink active={false}>Hello I am a link!</Header.TextLink>
                    </Header.Frame>
                </Header>
            </Router>
        );
    
        expect(getByText('Hello I am a link!')).toBeTruthy();
        expect(queryByTestId('header-bg')).toBeFalsy();
        // expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the full <Header> with a background', () => {
        const { container, getByText, getByTestId } = render(
            <Router>
                <Header src="joker1" dontShowOnSmallViewPort>
                    <Header.Frame>
                        <Header.Group>
                            <Header.Logo to={ROUTES.HOME} src="/images/logo.svg" alt="Netflix" />
                            <Header.TextLink active={false} onClick={() => {}}>
                                Series
                            </Header.TextLink>
                            <Header.TextLink active onClick={() => {}}>
                                Films
                            </Header.TextLink>
                        </Header.Group>
                        <Header.Group>
                            <Header.Search searchTerm="Joker" setSearchTerm={() => {}} />
                            <Header.Profile>
                                <Header.Picture src="/images/karl.png" />
                                <Header.Dropdown>
                                    <Header.Group>
                                        <Header.Picture src="/images/karl.png" />
                                        <Header.TextLink>Karl Hadwen</Header.TextLink>
                                    </Header.Group>
                                    <Header.Group>
                                        <Header.TextLink onClick={() => {}}>Sign out</Header.TextLink>
                                    </Header.Group>
                                </Header.Dropdown>
                            </Header.Profile>
                        </Header.Group>
                    </Header.Frame>
            
                    <Header.Feature>
                        <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
                        <Header.Text>Forever alone in a crowd...</Header.Text>
                        <Header.PlayButton>Play</Header.PlayButton>
                    </Header.Feature>
                </Header>
            </Router>
        );

        expect(getByTestId('search-input')).toBeTruthy();
        expect(getByTestId('search-input').value).toBe('Joker');
        fireEvent.change(getByTestId('search-input'), { target: { value: 'Simpsons' } });
        fireEvent.click(getByTestId('search-click'));

        expect(getByText('Series')).toBeTruthy();
        expect(getByText('Films')).toBeTruthy();
        expect(getByText('Karl Hadwen')).toBeTruthy();
        expect(getByText('Watch Joker Now')).toBeTruthy();
        expect(getByText('Sign out')).toBeTruthy();
        expect(getByText('Forever alone in a crowd...')).toBeTruthy();
        expect(getByText('Play')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
    
});

