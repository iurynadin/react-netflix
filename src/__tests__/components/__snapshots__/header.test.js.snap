// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the basic <Header /> with a background 1`] = `
<div
  class="header__Background-djltcn-0 dnvocD"
  data-testid="header-bg"
>
  <div
    class="header__Container-djltcn-1 emmllS"
  >
    <a
      href=""
    >
      <img
        alt="Netflix"
        class="header__Logo-djltcn-7 ekjEUf"
        src="/logo.svg"
      />
    </a>
    <p
      class="header__Link-djltcn-2 hFmGTm"
    >
      Hello I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the full <Header> with a background 1`] = `
<div
  class="header__Background-djltcn-0 kAFPvB"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-djltcn-1 emmllS"
  >
    <div
      class="header__Group-djltcn-3 imRSHv"
    >
      <a
        href="/"
      >
        <img
          alt="Netflix"
          class="header__Logo-djltcn-7 ekjEUf"
          src="/images/logo.svg"
        />
      </a>
      <p
        class="header__Link-djltcn-2 kBuiOu"
      >
        Series
      </p>
      <p
        class="header__Link-djltcn-2 kBuiOu"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-djltcn-3 imRSHv"
    >
      <div
        class="header__Search-djltcn-5 bLKtPv"
      >
        <button
          class="header__SearchIcon-djltcn-6 efPWCS"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-djltcn-4 fErfWg"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-djltcn-12 bMyGYo"
      >
        <button
          class="header__Picture-djltcn-10 dEPaKe"
          src="/images/users//images/karl.png.png"
        />
        <div
          class="header__Dropdown-djltcn-11 gvfPSY"
        >
          <div
            class="header__Group-djltcn-3 imRSHv"
          >
            <button
              class="header__Picture-djltcn-10 dEPaKe"
              src="/images/users//images/karl.png.png"
            />
            <p
              class="header__Link-djltcn-2 kBuiOu"
            >
              Karl Hadwen
            </p>
          </div>
          <div
            class="header__Group-djltcn-3 imRSHv"
          >
            <p
              class="header__Link-djltcn-2 kBuiOu"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-djltcn-1 header__Feature-djltcn-9 emmllS juLpVH"
  >
    <h2
      class="header__FeatureCallOut-djltcn-13 cqqilK"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-djltcn-14 hPbYGf"
    >
      Forever alone in a crowd...
    </p>
    <button
      class="header__PlayButton-djltcn-15 enoKOF"
    >
      Play
    </button>
  </div>
</div>
`;
